porespy/generators/__imgen__.py

```diff
@@ -974,30 +975,157 @@ def cylinders(shape: List[int], radius: int, ncylinders: int,
+        90 meaning they are randomly rotated about the Z axis by as much
+        as +/- 90 degrees.
+    length : scalar
+        The length of the cylinders to add.  If ``None`` (default) then the
+        cylinders will extend beyond the domain in both directions so no ends
+        will exist. If a scalar value is given it will be interpreted as the
+        Euclidean distance between the two ends of the cylinder.  Note that
+        one or both of the ends *may* still lie outside the domain, depending
+        on the randomly chosen center point of the cylinder.
+    max_iter : scalar
+        The number of fractional fiber insertions used to target the requested
+        porosity. By default a value of 3 is used (and this is typically
+        effective in getting very close to the targeted porosity), but a
+        greater number can be input to improve the achieved porosity.
+    return_fiber_number : bool
+        Determines whether the function will return the number of fibers
+        along with the image
+
+    Returns
+    -------
+    image : ND-array
+        A boolean array with ``True`` values denoting the pore space
+    fiber_number : scalar
+        The number of fibers added to the image. This is optional, and whether
+        it is returned is determined by 'return_fiber_number'
+    """
+    if max_iter < 3:
+        raise Exception("Iterations must be greater than or equal to 3")
+
+    vol_total = float(np.prod(shape))
+
+    def get_num_pixels(porosity):
+        r"""
+        Helper method to calculate number of pixels given a porosity
+        """
+        return -np.log(porosity) * vol_total
+
+    # Crudely estimate fiber length as cube root of product of dims
+    length_estimate = vol_total ** (1 / 3) if length is None else length
+
+    # Rough fiber volume estimate
+    vol_fiber = length_estimate * np.pi * radius * radius
+    n_pixels_to_add = get_num_pixels(porosity)
+
+    # Rough estimate of n_fibers
+    n_fibers_added = 0
+    # Calculate fraction of fibers to be added in each iteration.
+    subdif = 0.8 / np.sum(np.arange(1, max_iter) ** 2)
+    fractions = [0.2]
+    for i in range(1, max_iter):
+        fractions.append(fractions[i - 1] + (max_iter - i) ** 2 * subdif)
+
+    im = np.ones(shape, dtype=bool)
+    for frac in tqdm(fractions, disable=not verbose, file=sys.stdout,
+                     desc="Adding fibers"):
+        n_fibers_total = n_pixels_to_add / vol_fiber
+        n_fibers = int(np.ceil(frac * n_fibers_total) - n_fibers_added)
+        if n_fibers > 0:
+            im = im & ps.generators.cylinders(
+                shape, radius, n_fibers, phi_max, theta_max, length, verbose=False
+            )
+        n_fibers_added += n_fibers
+        # Update parameters for next iteration
+        porosity = ps.metrics.porosity(im)
+        vol_added = get_num_pixels(porosity)
+        vol_fiber = vol_added / n_fibers_added
+
+    if verbose:
```

Comment:
Regarding my above comment about the module-wide settings, we should also consider using the logger for these printed messages, so they can be turned off too, but setting loglevel very high.  In any case, remove verbose from here, so it always prints.  This is currently the normal porespy behavior.